### DISCUSSION TITLE
How to fetch fields values in graphql?

### DISCUSSION THREAD

**Discussion by vitalijalbu** [craftcms/cms#12020] [Q&A] [answered]
Hi guys, I have a huge list of **checkboxes** in a field and want to fetch all these fields values into my headless react app. How to do it with graphql?
As I see _craftcms graphql doesn't have access globally to fields_, but only entries.

**Comment by brandonkelly** [comment, marked as answer]
It’s only possible to fetch the selected checkbox values, not all of the options. You will need to hardcode the options in your React app.

**Reply by vitalijalbu** [reply]
Where I can copy some php code to do that? At leas as rest-api values... thanks

**Reply by brandonkelly** [reply]
You can fetch the field and its options from a module like so:

```php
/** @var \craft\fields\Checkboxes $field */
$field = Craft::$app->fields->getFieldByHandle('myFieldHandle');

// From here, you can get the options via $field->options.
// It will be set to an array where each item is a sub-array with `label` and `value` keys.
```
